### PR TITLE
feat: introducing playwright tests

### LINKED ISSUE #409: Introduce playwright tests and Chromatic to all General Vis visualization types and settings
Use the recently introduced `breastCancerData` dataset to construct visualizations and test settings permutations, label changes, configuration properties, etc.

### PR DESCRIPTION
**Pull request by dvlilialbrecht**
Closes #409 

Creating playwright tests for all visyn core plots (scatter plot, box plot, correlation plot, heat map, hexbin, violin plot). All plots have multiple test-files, with general tests (download, selection etc.) and with every setting.

Tests were created with chromatic, so all visual changes can be viewed [there](https://www.chromatic.com/builds?appId=640e9afaf15e817c63e10f42&branch=lalbrecht%2F409_introducing-playwright-tests).

Still open issue: The Scatter plot problem. Chromatic does not show dots of WebGL plot. This is most likely a Chromatic issue, since when taking screenshots with playwright, these dots are visible. 
Screenshot from 2024-08-21 10-55-56

Tests were also run in the CI: Timed out waiting 600s for the test suite to run
Following tests are commended out: 
- boxPlot / 02_numericalColumns / one numerical column selected
- boxPlot / 02_numericalColumns / two numerical column selected
- scatterPlot / 01_general / download plot
- scatterPlot / 08_regressionLine / should not be searchable
- violinPlot / 01_general / download plot
- violinPlot / 03_categoricalColumns / no categorical column selected

### CODE REVIEW

File: .yarnrc.yml [hunk 1]
```diff
@@ -1,3 +1,7 @@
+compressionLevel: mixed
```
Comment (puehringer) [inline .yarnrc.yml:1]: Revert this change

**Comment by puehringer** [comment]
@dvlilialbrecht @dvdanielamoitzi please update the build.yml and build_nightly.yml from Aevidence.

File: src/vis/EagerVis.tsx [hunk 2]
```diff
@@ -220,7 +220,7 @@ export function EagerVis({
 
   // eslint-disable-next-line @typescript-eslint/naming-convention
   const [_visConfig, _setVisConfig] = useUncontrolled({
-    ...(isControlled ? { value: externalConfig, onChange: setExternalConfig } : {}),
+    ...(isControlled ? { value: wrapWithDefaults(externalConfig), onChange: setExternalConfig } : {}),
```
Comment (puehringer) [inline src/vis/EagerVis.tsx:223]: Undo

File: chromatic-diagnostics.json [hunk 3]
```diff
@@ -0,0 +1,861 @@
+{
```
Comment (puehringer) [inline chromatic-diagnostics.json:1]: Remove

File: package.json [hunk 4]
```diff
@@ -120,10 +122,33 @@
...
-    "chromatic": "^11.5.3",
+    "@chromatic-com/playwright": "^0.6.18",
+    "@chromatic-com/storybook": "^1.6.1",
+    "@playwright/test": "^1.45.2",
+    "@types/node": "^20.14.10",
+    "chromatic": "^11.6.0",
     "cypress": "^13.11.0"
   },
+  "resolutions": {
+    "storybook": "8.2.6",
```
Comment (puehringer) [inline package.json:133]: @puehringer remove storybook from visyn_scripts and let the apps install it as they wish.

**Comment by puehringer** [comment]
@dvlilialbrecht @dvdanielamoitzi I updated everything to visyn_scripts 11 and removed the storybook overrides, and let the build run: https://github.com/datavisyn/visyn_core/actions/runs/10698246357/job/29657200985

One thing I noticed: all our stories are using `import { ComponentStory } from '@storybook/react';`, but that only exists in `@storybook/react: 7`, not `8`. I thought chromatic installs storybook 8? Why doesn't it need `@storybook/react`? And now we are installing version 7 of `@storybook/react`, and 8 of storybook via chromatic? Sounds like a problem to me, so maybe we should go ahead and upgrade the stories to 8 as well? Let me know what you think.

**Review by dvdanielamoitzi — APPROVED** [review]
